cmake/modules/extensions.cmake

```diff
@@ -915,7 +917,9 @@ function(board_check_revision)
     endif()
   endif()
 
-  if(BOARD_REV_FORMAT STREQUAL LETTER)
+  if(BOARD_REV_FORMAT STREQUAL ALPHANUM)
+    set(revision_regex "([A-Z][0-9]+)")
```

Comment:
this `ALPHANUM` revision format seems to specific for the nucleo boards.

So far i'm not convinced we should extend the common helper for this case.

What about a two letters + <n> numbers, or the opposite, number, like `25hdd` ?
Should that be called `NUMALPHA`.

Note, boards can use any revision format they like, but not everything need to have built-in support in the `board_check_revision()` function.

For example, the legend board uses revisions `25hdd`, `25ssd`, `35`.
Ref:
https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/arm/legend/revision.cmake

Also read:
https://docs.zephyrproject.org/latest/hardware/porting/board_porting.html#custom-revision-cmake-files